src/component/mxgraph/BpmnMxGraph.ts

```diff
@@ -74,23 +81,69 @@ export class BpmnMxGraph extends mxGraph {
     return Math.max(input || 0, 0);
   }
 
-  // solution inspired by https://github.com/algenty/grafana-flowcharting/blob/0.9.0/src/graph_class.ts#L1254
-  public performZoom(up: boolean, evt: MouseEvent): void {
-    const rect = this.container.getBoundingClientRect();
-    const x = evt.clientX - rect.left;
-    const y = evt.clientY - rect.top;
-    this.zoomTo(null, null, up, x, y);
-  }
-
-  zoomTo(scale: number, center?: boolean, up?: boolean, offsetX?: number, offsetY?: number): void {
+  zoomTo(scale: number, center?: boolean, up?: boolean, offsetX?: number, offsetY?: number, performScaling?: boolean): void {
     if (scale === null) {
       const [newScale, dx, dy] = this.getScaleAndTranslationDeltas(up, offsetX, offsetY);
-      this.view.scaleAndTranslate(newScale, this.view.translate.x + dx, this.view.translate.y + dy);
+      if (performScaling) {
+        this.view.scaleAndTranslate(newScale, this.view.translate.x + dx, this.view.translate.y + dy);
+      }
     } else {
       super.zoomTo(scale, center);
     }
   }
 
+  createMouseWheelZoomExperience(config: ZoomConfiguration): void {
+    config = this.ensureValidZoomConfiguration(config);
+    mxEvent.addMouseWheelListener(debounce(this.getZoomHandler(true), config.debounceDelay), this.container);
+    mxEvent.addMouseWheelListener(throttle(this.getZoomHandler(false), config.throttleDelay), this.container);
+  }
+
+  /**
+   * Make sure the configuration parameters are defined and valid (range 0 .. 100), if not assign defaults
+   * @param config ZoomConfiguration
+   */
+  private ensureValidZoomConfiguration(config: ZoomConfiguration): ZoomConfiguration {
+    const validatedConfig = config || { debounceDelay: 50, throttleDelay: 50 };
+    if (!config.debounceDelay && (config.debounceDelay < 0 || config.debounceDelay > 100)) {
+      validatedConfig.debounceDelay = 50;
+    }
+    if (!config.throttleDelay && (config.throttleDelay < 0 || config.throttleDelay > 100)) {
+      validatedConfig.throttleDelay = 50;
+    }
+    return validatedConfig;
```

Comment:
:warning: duplication and could be unit tested. I will propose something

Reply:
see 4a35cf2

Reply:
ahh, missed part 'I will propose something'
got something similar:
![image](https://user-images.githubusercontent.com/45601541/98649697-2a50a100-2338-11eb-818b-ff02317eda51.png)